pkg/apis/ceph.rook.io/v1/network.go

```diff
@@ -32,10 +32,23 @@ func (n *NetworkSpec) IsMultus() bool {
 	return n.Provider == NetworkProviderMultus
 }
 
-// IsHost get whether to use host network provider. This method also preserve
-// compatibility with the old HostNetwork field.
-func (n *NetworkSpec) IsHost() bool {
-	return (n.HostNetwork && n.Provider == NetworkProviderDefault) || n.Provider == NetworkProviderHost
+// IsHost is intended to be used to determine if the rook operator should configure
+// managed pods to use host networking.
+// This behavior is enabled by configuring the cephCluster with the "host" network provider.
+// This method also maintains compatibility with the old HostNetwork setting
+// which is incompatible with other network providers: HostNetwork set to true
+// together with an empty or unset network provider has the same effect as
+// network.Provider set to "host"
+func (n *NetworkSpec) IsHost() (ret bool, err error) {
```

Comment:
I see there a lot of bugs with this implementation 
```
func (n *NetworkSpec) IsHost() (bool, error) {
	ret := false
	err := ValidateNetworkSpec("", *n)
	if err != nil {
		return ret, errors.Wrapf(err, "Unable to determine  host network setting from Invalid  network spec. not setting host network.")
	}
	return (n.HostNetwork && n.Provider == NetworkProviderDefault) || n.Provider == NetworkProviderHost, err
}
```

Fixing this make the ` make build` worked

Reply:
@parth-gr   this has been fixed.